.github/workflows/README.md

```diff
@@ -35,27 +68,27 @@ The CD Workflow is responsible for automatically deploying new code changes to d
 
 
 
-**Workflow Trigger and Jobs**
-
-The CD Workflow is triggered by manual trigger through workflow dispatch, allowing developers to initiate the deployment process on demand.
-
-The CD Workflow consists of several jobs, each responsible for a specific deployment task. For instance, the `deploy-to-dev` job deploys the dev build to Firebase, while the `promote-artifact` job promotes the release build to the specified Google Play track.
-
-## **Dependency Updates workflow**
+**Version Code**
 
-Updates project dependencies using Dependabot, an automated dependency management tool, ensuring that the project always uses the latest stable versions of its dependencies.
+The version code is generated from 3 things:
+1. A "base" version code of `10000000`. For a time the version code was derived from unix time / 1000. This created a always increasing number, but was otherwise not so useful. Setting a base allows us to establish a clean "floor" for the version code. If the workflow runs ever exceed the base, first of all go us, and 2nd the base can be removed. 
+2. The `run number` of the workflow. This tells us which run a build came from and is auto incrementing. 
+3. The last two digits are reserved for the `run attempts`. This tells us how many times a specific workflow was run. If more than 99 attempts are made (something is wrong) the build will fail and you need to start a new workflow run. 
 
-## **Reusable Workflows**
+- Ex: `100001502` means this build came from workflow 15, run 2. 
 
-To promote code reusability and efficiency, two reusable workflows are defined:
+**Version Name**
 
-1.  **Test Android Modules:** This workflow takes a list of modules and a unique report ID as input and executes unit tests for the specified modules. It then uploads the test coverage reports using the provided report ID.
+The version name follows our versioning convention:
+- `year`.`quarter`.`release`-`(optional) deployment`+`(optional) unix timestamp`.`run number`.`run attempt`
 
-2.  **Deploy to Firebase:** This workflow takes a build type (dev or staging) as input and uploads the corresponding APK (debug or staging) to the Firebase distribution track.
+The optional params are **only** used on none release builds. We add the unix timestamp so we can compute the `versionCode` from the filename if needed.  
```

Comment:
That's no longer valid, right? 
- `year`.`quarter`.`release`-`(optional) deployment`+`(10000000+run number+run attempt)`

Reply:
The unix date is no longer valid good point! But also we don't currently put the "base" number in the version name, only the version code. 

Version name = 2024.1.0+15.2
Version code = 100001502

Since the version name is user and delivery facing I thought `2024.1.0+15.2` was more friendly than `2024.1.0+100001502`. You can still compute the version code from the version name if we need to.